staging/src/k8s.io/apiserver/pkg/admission/plugin/resourcequota/controller.go

```diff
@@ -567,6 +568,14 @@ func CheckRequest(quotas []corev1.ResourceQuota, a admission.Attributes, evaluat
 					prettyPrint(failedHard)))
 		}
 
+		// use the same format in used as in hard to prevent inconsistent unit format switching
+		for resourceName, usedQuantity := range newUsage {
+			hardQuantity := resourceQuota.Status.Hard[resourceName]
+			if hardQuantity.Format != usedQuantity.Format {
+				newUsage[resourceName] = *resource.NewQuantity(usedQuantity.Value(), hardQuantity.Format)
+			}
+		}
+
```

Comment:
the change above may not be persisted, it will be skipped if there is no change

https://github.com/kubernetes/kubernetes/blob/4e443b410091080263104e82bac523a13ebb9796/staging/src/k8s.io/apiserver/pkg/admission/plugin/resourcequota/controller.go#L263

Reply:
I think that these two use cases should cover all the places that change status values. In that case every time the value changes (dirty == true), the format should change too. Please correct me if I am wrong. I would prefer to have less places which set dirty to prevent unnecessary updates.

Also in case we would like to set dirty flag, the logic would have to be a bit more complex.